packages/core/rollup.config.js

```diff
@@ -48,9 +49,11 @@ const createUmdConfig = ({ input, output, target = undefined }) => ({
   ]
 });
 
+const npmInputs = glob.sync('src/!(scxml|invoke|model.types|typegenTypes).ts');
+
 export default [
   createNpmConfig({
-    input: 'src/index.ts',
+    input: npmInputs,
```

Comment:
Even though we were using `preserveModules` here the whole thing was still subject to tree-shaking and since `keys` was not really used by anything... it got removed by Rollup. By providing an array of inputs we let Rollup to know that we expect stuff to be importable directly from those entries so it has to preserve every public export of every input - and that makes `keys` to be preserved